### PR TITLE
Support kinds of emitter format-setting for YAML::Null.

### PR DESCRIPTION
**Pull request by dota17**
Add four emitter styles for YAML::Null,  like  #137 
```C++
  LowerNull   // null
  UpperNull   // NULL
  CamelNull   // Null
  TildeNull   // ~
```

### CODE REVIEW

File: src/emitter.cpp [hunk 1]
```diff
@@ -893,7 +912,8 @@ Emitter& Emitter::Write(const _Null& /*null*/) {
 
   PrepareNode(EmitterNodeType::Scalar);
 
-  m_stream << "~";
+  const char* name = ComputeNullName();
+  m_stream << name;
```
Comment (jbeder) [inline src/emitter.cpp:916]: Just inline the function call.

File: src/emitter.cpp [hunk 2]
```diff
@@ -770,6 +774,21 @@ const char* Emitter::ComputeFullBoolName(bool b) const {
                          // these answers
 }
 
+const char* Emitter::ComputeNullName() const {
+  const EMITTER_MANIP nullFmt = m_pState->GetNullFormat();
```
Comment (jbeder) [inline src/emitter.cpp:778]: Inline this in the switch (*) statement.

File: src/emitter.cpp [hunk 3]
```diff
@@ -770,6 +774,21 @@ const char* Emitter::ComputeFullBoolName(bool b) const {
...
+  switch (nullFmt) {
+    case LowerNull:
+      return "null";
+    case UpperNull:
+      return "NULL";
+    case CamelNull:
+      return "Null";
+    case TildeNull:
+    default:
```
Comment (jbeder) [inline src/emitter.cpp:787]: Explicitly write

// fallthrough

File: test/integration/emitter_test.cpp [hunk 4]
```diff
@@ -1047,6 +1047,37 @@ TEST_F(EmitterTest, BoolFormatting) {
...
+  out.SetNullFormat(LowerNull);
+  out << Null;
+  out.SetNullFormat(UpperNull);
+  out << Null;
+  out.SetNullFormat(CamelNull);
+  out << Null;
+  out.SetNullFormat(TildeNull);
+  out << Null;
+  out << EndSeq;
+  ExpectEmit("- null\n- NULL\n- Null\n- ~");
```
Comment (jbeder) [inline test/integration/emitter_test.cpp:1061]: I know the related tests don't do this, but I think it'd be easier to read if this sequence is written in flow style (and below).

File: test/integration/emitter_test.cpp [hunk 5]
```diff
@@ -1047,6 +1047,37 @@ TEST_F(EmitterTest, BoolFormatting) {
...
+  out << LowerNull << Null;
+  out << UpperNull << Null;
+  out << CamelNull << Null;
+  out << TildeNull << Null;
+  out << EndSeq;
+  ExpectEmit("- null\n- NULL\n- Null\n- ~");
+}
+
+TEST_F(EmitterTest, NullFormattingOnNode) {
+  Node n(Load("null"));
```
Comment (jbeder) [inline test/integration/emitter_test.cpp:1075]: Can you run through all four cases here too?

File: src/emitter.cpp [hunk 6]
```diff
@@ -770,6 +774,21 @@ const char* Emitter::ComputeFullBoolName(bool b) const {
...
+  switch (m_pState->GetNullFormat()) {
+    case LowerNull:
+      return "null";
+    case UpperNull:
+      return "NULL";
+    case CamelNull:
+      return "Null";
+    case TildeNull:
+      return "~";
+    default:  // fallthrough
```
Comment (jbeder) [inline src/emitter.cpp:787]: What I meant was:

```
case TildeNull:
  // fallthrough
default:
  return "~";
```

This is the same as the previous behavior you had, but with the comment.

What you have now would not return any value in the default case, which would be undefined behavior.

Comment (dota17) [reply src/emitter.cpp]: haha, I misunderstanded your meaning.

**Pull request merged**